packages/mobile/src/send/SendAmount.tsx

```diff
@@ -1,254 +1,217 @@
-import CurrencyDisplay, { FormatType } from 'src/components/CurrencyDisplay'
 import {
   DAILY_PAYMENT_LIMIT_CUSD,
   DOLLAR_TRANSACTION_MIN_AMOUNT,
-  MAX_COMMENT_LENGTH,
   NUMBER_INPUT_MAX_DECIMALS,
 } from 'src/config'
-import { FeeType } from 'src/fees/actions'
-import EstimateFee from 'src/fees/EstimateFee'
 import { getFeeEstimateDollars } from 'src/fees/selectors'
-import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
-import i18n, { Namespaces, withTranslation } from 'src/i18n'
+import { Namespaces } from 'src/i18n'
 import { fetchAddressesAndValidate } from 'src/identity/actions'
-import { AddressValidationType, RecipientVerificationStatus } from 'src/identity/reducer'
+import {
+  AddressValidationType,
+  e164NumberToAddressSelector,
+  RecipientVerificationStatus,
+  secureSendPhoneNumberMappingSelector,
+} from 'src/identity/reducer'
 import { getAddressValidationType } from 'src/identity/secureSend'
-import { LocalCurrencyCode, LocalCurrencySymbol } from 'src/localCurrency/consts'
 import {
   convertDollarsToLocalAmount,
   convertDollarsToMaxSupportedPrecision,
   convertLocalAmountToDollars,
 } from 'src/localCurrency/convert'
-import { getLocalCurrencyCode, getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
-import { HeaderTitleWithBalance, headerWithBackButton } from 'src/navigator/Headers'
+import {
+  getLocalCurrencyCode,
+  getLocalCurrencyExchangeRate,
+  getLocalCurrencySymbol,
+} from 'src/localCurrency/selectors'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 import { getRecipientVerificationStatus, Recipient } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { PaymentInfo } from 'src/send/reducers'
+import useSelector from 'src/redux/useSelector'
 import { getRecentPayments } from 'src/send/selectors'
-import { dailyAmountRemaining, getFeeType } from 'src/send/utils'
+import { dailyAmountRemaining, getFeeType, isPaymentLimitReached } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import { fetchDollarBalance } from 'src/stableToken/actions'
-import { withDecimalSeparator } from 'src/utils/withDecimalSeparator'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
 
 export interface TransactionDataInput {
   recipient: Recipient
   amount: BigNumber
-  reason: string
   type: TokenTransactionType
+  reason: string
   firebasePendingRequestUid?: string | null
 }
 
-const AmountInput = withDecimalSeparator(
-  withTextInputLabeling<ValidatedTextInputProps<DecimalValidatorProps>>(ValidatedTextInput)
-)
-const CommentInput = withTextInputLabeling<TextInputProps>(TextInput)
+type RouteProps = StackScreenProps<StackParamList, Screens.SendAmount>
+type Props = RouteProps
 
-interface State {
-  amount: string
-  reason: string
-}
+const { decimalSeparator } = getNumberFormatSettings()
```

Comment:
Yes, but the app needs to be restarted. Essentially, even if it was not restarted - it would show previous `decimalSeparator`, but would be consistent around the app.